pkg/restapi/backup.go

```diff
@@ -79,7 +79,7 @@ func (h backupHandler) locationsCtx(next http.Handler) http.Handler {
 }
 
 func (h backupHandler) extractLocations(r *http.Request) ([]backupspec.Location, error) {
-	tasks, err := h.schedSvc.ListTasks(r.Context(), mustClusterIDFromCtx(r), scheduler.ListFilter{TaskType: []scheduler.TaskType{scheduler.BackupTask}})
+	tasks, err := h.schedSvc.ListTasks(r.Context(), mustClusterIDFromCtx(r), scheduler.ListFilter{TaskType: []scheduler.TaskType{scheduler.BackupTask}, Deleted: true})
```

Comment:
We should probably not extract locations from deleted tasks.
A location is valid provided that there is at least one backup using it.